src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6145,15 +6079,17 @@ void VM_Crac::doit() {
         print_resources("OK: jcmd socket\n");
         continue;
       }
-      details = sock_details(details, detailsbuf, sizeof(detailsbuf));
-      print_resources(" details2=\"%s\" ", details);
     }
 
     print_resources("BAD: opened by application\n");
     ok = false;
 
-    char* msg = NEW_C_HEAP_ARRAY(char, strlen(details) + 1, mtInternal);
-    strcpy(msg, details);
+    const int maxinfo = 64;
+    size_t buflen = strlen(details) + maxinfo;
+    char* msg = NEW_C_HEAP_ARRAY(char, buflen, mtInternal);
+    int ilen = snprintf(msg, maxinfo, "FD fd=%d type=%s path=", i, type);
+    ilen = ilen > maxinfo ? maxinfo : ilen;
+    strncpy(msg + ilen, detailsbuf, buflen - ilen);
```

Comment:
`ilen >= maxinfo` if the output was truncated [1]

`strncpy` may leave the string unterminated if `buflen-ilen` smaller than details.

So `snprintf(msg, maxinfo, "FD fd=%d type=%s path=%s", i, type, details)` will be a safer option. 

[1] RETURN in https://linux.die.net/man/3/snprintf